applications/madini_tracechain.md

```diff
@@ -129,37 +170,75 @@ Project Details
 
 We expect the teams to already have a solid idea about the project's expected final state. Therefore, we ask the teams to submit (where relevant):
 
-    Mockups/designs of any UI components
-    API specifications of the core functionality
-    An overview of the technology stack to be used
-    Documentation of core components, protocols, architecture, etc. to be deployed
-    PoC/MVP or other relevant prior work or research on the topic
```

Comment:
Sample Code using VIP 191 To send a transaction to Vechain Blockchain.

@auth.route('/createnew', methods=['GET', 'POST'])
def CreatenewRouted():
if request.method == "POST":
hashdata = str(request.json['hashdata'])
des = str(request.json['des'])
sku = str(request.json['sku'])
pname = str(request.json['pname'])
ptype = str(request.json['type'])
mainimage = str(request.json['mainimage'])
_node_url = node
BlockInfos = requests.get(_node_url + '/blocks/best') 
_BlockRef = BlockInfos.json()['id'][0:18]
_Nonce = randint(10000000, 99999999)
delegated_body = {
"chainTag": chaintag,
"blockRef": _BlockRef,
"expiration": 720,
"clauses": [
{
"to": '',#enter to address
"value": 0,
"data": '0x' +username
}
],
"gasPriceCoef": 0,
"gas": 100000,
"dependsOn": None,
"nonce": _Nonce,
"reserved": {
"features": 1
}
}
print(hashdata)
delegated_tx = transaction.Transaction(delegated_body)
assert delegated_tx.is_delegated() == True
# Sender address
addr_1 = ''
priv_1 = bytes.fromhex('') # Sender private key address
# Gas 5
addr_2 = '' # Gas Payer
priv_2 = bytes.fromhex('')#Gas payer key
h = delegated_tx.get_signing_hash() # Sender hash to be signed.
dh = delegated_tx.get_signing_hash(addr_1) # Gas Payer hash to be signed.
# Sender sign the hash.
# Gas payer sign the hash.
# Concat two parts to forge a legal signature.
sig = cry.secp256k1.sign(h, priv_1) + cry.secp256k1.sign(dh, priv_2)
delegated_tx.set_signature(sig)
ts = datetime.datetime.now()
tx_headers = {'Content-Type': 'application/json', 'accept': 'application/json'}
send_transaction = requests.post(_node_url + '/transactions', json=tx_data, headers=tx_headers)
print('Response from Server: ' + str(send_transaction.content))
prt=str(send_transaction.content)
encoded_bytes = delegated_tx.encode()
new_verif = Verif(bhash=des, mid=username, thash=delegated_tx.get_id(), sku=sku, pname=pname ,mainimage=mainimage ,datetime=ts) #
db.session.add(new_verif)
db.session.commit()
return "<a target='blank' href='https://explore.vechain.org/transactions/"+delegated_tx.get_id()+"'>Explorer</a>"
encoded_bytes = delegated_tx.encode()
tx_data = {'raw': '0x' + encoded_bytes.hex()}
print(tx_data)
ts = datetime.datetime.now()
tx_headers = {'Content-Type': 'application/json', 'accept': 'application/json'}
send_transaction = requests.post(_node_url + '/transactions', json=tx_data, headers=tx_headers)
print('Response from Server: ' + str(send_transaction.content))
prt=str(send_transaction.content)
encoded_bytes = delegated_tx.encode()
new_verif = Verif(bhash=des, mid=username, thash=delegated_tx.get_id(), sku=sku, pname=pname ,mainimage=mainimage ,datetime=ts,ptype=ptype,shape=shape,color=color,carats=carats,clarity=clarity,cut=cut,category=category) #
db.session.add(new_verif)
db.session.commit()
return "<a target='blank' href='https://explore.vechain.org/transactions/"+delegated_tx.get_id()+"'>Explorer</a>"